### ISSUE TITLE
Support `Split` even when dimension is not evenly splittable

### ISSUE THREAD

**Issue by xuzijian629** [onnx/onnx#4438]
Currently, [Split](https://github.com/onnx/onnx/blob/v1.12.0/docs/Operators.md#split) operator splits the tensor to **equal sized** tensors if `split` input is not given.

I think this restriction is not very important. ~For example, pytorch's split also supports the case when dim cannot be evenly split. https://pytorch.org/docs/1.12/generated/torch.split.html?highlight=split#torch.split~

UPD: Torch's corresponding function was https://pytorch.org/docs/stable/generated/torch.tensor_split.html

Can we update the operator definition to support this case?

**Comment by gramalingam** [comment]
Sounds reasonable to me.

**Comment by xuzijian629** [comment]
Should I implement this as Split-18 at the same time with https://github.com/onnx/onnx/pull/4389, or is it better to implement this as Split-19?

**Comment by gramalingam** [comment]
Adding @p-wysocki, who is working on Split-18 ... I think if everyone agrees on the extension, it would be better to have it together in Split-18 itself (if possible; otherwise, we can always do Split-19 as a fallback).

**Comment by p-wysocki** [comment]
Sounds good, I'll add it to Split-18.

**Comment by p-wysocki** [comment]
PR: https://github.com/onnx/onnx/pull/4481

**Comment by p-wysocki** [comment]
The PR has been merged. Please reopen the issue if needed.
